libgadget/blackhole.c

```diff
@@ -515,12 +653,60 @@ blackhole_accretion_preprocess(int n, TreeWalk * tw)
 static void
 blackhole_feedback_postprocess(int n, TreeWalk * tw)
 {
+    int j;
     int PI = P[n].PI;
+
     if(BH_GET_PRIV(tw)->BH_accreted_Mass[PI] > 0)
     {
         P[n].Mass += BH_GET_PRIV(tw)->BH_accreted_Mass[PI];
         BHP(n).Mass += BH_GET_PRIV(tw)->BH_accreted_BHMass[PI];
     }
+
+    /*******************************************************************/
+    /* Add accel. except for reposition case */
+    if(All.BH_DynamicalFriction < 3){ 
+        double bhvel = 0;
+        double c_over_sqrt2;
+        double bmax, bmin;
+        double log_lam, rho_frac;
+
+        /* Calculate Coulumb Logarithm */
+        c_over_sqrt2 = LIGHTCGS * All.UnitVelocity_in_cm_per_s / sqrt(2.);
+
+        for(j = 0; j < 3; j++) 
+        {
+            bhvel += pow(P[n].Vel[j] - BH_GET_PRIV(tw)->BH_SurroundingVel[PI][j], 2);
+        }
+        bhvel = sqrt(bhvel);
+
+        if (bhvel > c_over_sqrt2)
+            bmin = 2. * All.G * P[n].Mass / pow(LIGHTCGS * All.UnitVelocity_in_cm_per_s, 2);
+        else
+            bmin = All.G * P[n].Mass / pow(bhvel, 2);
+
+        bmax = All.FractionalGravitySoftening * All.MeanSeparation[1]; 
+        log_lam = log(bmax/bmin);
+        rho_frac =   BH_GET_PRIV(tw)->BH_DFFracMass[PI] / BH_GET_PRIV(tw)->BH_DFAllMass[PI] * BH_GET_PRIV(tw)->BH_SurroundingDensity[PI];
+
+        /* Simplified Version from Tremmel 2015 Eq. (3) */
+        for(j = 0; j < 3; j++) 
+        {
+            BHP(n).DFAccel[j] = - 4. * M_PI * All.G * All.G * P[n].Mass * rho_frac * 
+            log_lam * (P[n].Vel[j] - BH_GET_PRIV(tw)->BH_SurroundingVel[PI][j]) / pow(bhvel, 3);
+            P[n].GravAccel[j]  += All.BH_DFBoost * BHP(n).DFAccel[j]; // Add a boost factor
+        }
+    }
+    else
+    {
+        for(j = 0; j < 3; j++) 
+        {
```

Comment:
Better would be to just not allocate the array if DynamicalFriction is off and avoid this else clause. On use one would check for a NULL pointer in this array instead. An alternative is to change this to an unconditional memset(DFAccel, 0, 3) before the if clause.